torch/_inductor/ir.py

```diff
@@ -825,7 +825,19 @@ def fn(index):
             }
             and config.split_reductions
         )
-        if split_reduction and not dynamo_config.dynamic_shapes:
+        sr_qualified = split_reduction
+        if sr_qualified and dynamo_config.dynamic_shapes:
```

Comment:
Is there a reason to gate this on `dynamo_config.dynamic_shapes`? It seems like it should be OK to always do this test

Reply:
Yea, but it's not free, why do a bunch of checks for no reason?

Reply:
...because you're eventually going to turn on dynamic_shapes = True everywhere by default?